TrackletAlgorithm/TrackletCalculator.cc

```diff
@@ -28,7 +28,7 @@ void TrackletCalculator_L1L2E(
 #pragma HLS array_partition variable=projout_barrel_2s complete
 #pragma HLS array_partition variable=projout_disk complete
 
-  TC_L1L2E: TrackletCalculator<TC::L1L2, TC::E, BARRELPS, BARRELPS, 2, 1, 13, 0x1FE0, 0x0000, 0x7772000, 0x7777, kMaxProc>(
```

Comment:
these magic numbers make me nervous....

Reply:
These numbers come directly from the wiring, and I think ultimately, we want a script to generate these top functions automatically from the wires file. I can start working on that, but beyond that, I'm not sure what to do about these numbers.

Reply:
should they at least become named constants so people have a hope of understanding them?

Reply:
OK, I moved the magic numbers to a new file called `TrackletCalculator_parameters.h`, which, in the future, can be generated by a script from the wiring file. I decided to organize the numbers into constexpr template functions to reflect the fact that they are functions of the seed and iTC letter.

I also realized that the masks didn't need to be parameters of the top template function anymore, so I removed them.